### PR TITLE
Ensure consistent IRO for all objects.

### LINKED ISSUE #49: Objects have inconsistent resolution orders
Exposed with zope.interface 5.0. E.g.,

```
zope.interface.ro.InconsistentResolutionOrderError: InconsistentResolutionOrderError: For object <implementedBy zope.configuration.fields.GlobalInterface>.
Base ROs:
{<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>: [<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
                                                                  <InterfaceClass zope.interface.Interface>],
 <implementedBy zope.configuration.fields.GlobalObject>: [<implementedBy zope.configuration.fields.GlobalObject>,
                                                          <InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
                                                          <implementedBy zope.schema._bootstrapfields.Field>,
                                                          <InterfaceClass zope.schema.interfaces.IField>,
                                                          <InterfaceClass zope.schema._bootstrapinterfaces.IValidatable>,
                                                          <implementedBy zope.interface.interface.Attribute>,
                                                          <InterfaceClass zope.interface.interfaces.IAttribute>,
                                                          <implementedBy zope.interface.interface.Element>,
                                                          <InterfaceClass zope.interface.interfaces.IElement>,
                                                          <implementedBy builtins.object>,
                                                          <InterfaceClass zope.interface.Interface>]}
Conflict Location:
[[<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <InterfaceClass zope.interface.Interface>],
 [<implementedBy zope.configuration.fields.GlobalObject>,
  <InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <implementedBy zope.schema._bootstrapfields.Field>,
  <InterfaceClass zope.schema.interfaces.IField>,
  <InterfaceClass zope.schema._bootstrapinterfaces.IValidatable>,
  <implementedBy zope.interface.interface.Attribute>,
  <InterfaceClass zope.interface.interfaces.IAttribute>,
  <implementedBy zope.interface.interface.Element>,
  <InterfaceClass zope.interface.interfaces.IElement>,
  <implementedBy builtins.object>,
  <InterfaceClass zope.interface.Interface>],
 [<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <implementedBy zope.configuration.fields.GlobalObject>]]

InconsistentResolutionOrderError: For object <implementedBy zope.configuration.fields.Path>.
Base ROs:
{<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>: [<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
                                                                  <InterfaceClass zope.interface.Interface>],
 <implementedBy zope.schema._bootstrapfields.Text>: [<implementedBy zope.schema._bootstrapfields.Text>,
                                                     <InterfaceClass zope.schema.interfaces.IText>,
                                                     <InterfaceClass zope.schema.interfaces.IMinMaxLen>,
                                                     <InterfaceClass zope.schema.interfaces.ILen>,
                                                     <InterfaceClass zope.schema.interfaces.IIterable>,
                                                     <InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
                                                     <implementedBy zope.schema._bootstrapfields.MinMaxLen>,
                                                     <implementedBy zope.schema._bootstrapfields.Field>,
                                                     <InterfaceClass zope.schema.interfaces.IField>,
                                                     <InterfaceClass zope.schema._bootstrapinterfaces.IValidatable>,
                                                     <implementedBy zope.interface.interface.Attribute>,
                                                     <InterfaceClass zope.interface.interfaces.IAttribute>,
                                                     <implementedBy zope.interface.interface.Element>,
                                                     <InterfaceClass zope.interface.interfaces.IElement>,
                                                     <implementedBy builtins.object>,
                                                     <InterfaceClass zope.interface.Interface>]}
Conflict Location:
[[<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <InterfaceClass zope.interface.Interface>],
 [<implementedBy zope.schema._bootstrapfields.Text>,
  <InterfaceClass zope.schema.interfaces.IText>,
  <InterfaceClass zope.schema.interfaces.IMinMaxLen>,
  <InterfaceClass zope.schema.interfaces.ILen>,
  <InterfaceClass zope.schema.interfaces.IIterable>,
  <InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <implementedBy zope.schema._bootstrapfields.MinMaxLen>,
  <implementedBy zope.schema._bootstrapfields.Field>,
  <InterfaceClass zope.schema.interfaces.IField>,
  <InterfaceClass zope.schema._bootstrapinterfaces.IValidatable>,
  <implementedBy zope.interface.interface.Attribute>,
  <InterfaceClass zope.interface.interfaces.IAttribute>,
  <implementedBy zope.interface.interface.Element>,
  <InterfaceClass zope.interface.interfaces.IElement>,
  <implementedBy builtins.object>,
  <InterfaceClass zope.interface.Interface>],
 [<InterfaceClass zope.schema._bootstrapinterfaces.IFromUnicode>,
  <implementedBy zope.schema._bootstrapfields.Text>]]
```

### PR DESCRIPTION
**Pull request by jamadden**
This requires switching some tests to the plain unittest runner right now because of cyclic dependencies.

Fixes #49

Also drop 3.4 and add 3.8.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,12 +1,15 @@
 [tox]
 envlist =
-    py27,py34,py35,py36,py37,pypy,pypy3,coverage,docs
+    py27,py35,py36,py37,pypy,pypy3,coverage,docs
```
[the remark this answers is not in the document]
Comment (jamadden) [reply tox.ini]: Thanks.

**Comment by jamadden** [comment]
> I suspect the coverage drop that coveralls warns about is real.
> Is python -m unittest.discover running our doctests? 

I expected that; the switch to unittest over testrunner should only be temporary while the cycles get resolved.

**Comment by jamadden** [comment]
>> I suspect the coverage drop that coveralls warns about is real.
>> Is python -m unittest.discover running our doctests?
> I expected that; 

Ah, but `unittest` does not have an analogue to zope.testrunners `test_suite` method, `load_tests`, which serves our purpose here well enough.

**Comment by jamadden** [comment]
And back to 100%. Thank you!

**Comment by mgedmin** [comment]
> Ah, but unittest does not have an analogue to zope.testrunners test_suite method, load_tests, which serves our purpose here well enough.

does **not**?

**Comment by jamadden** [comment]
Supposed to be "no*w*"

**Pull request merged**